homeassistant/helpers/config_validation.py

```diff
@@ -89,14 +89,42 @@ def validate(obj: Dict) -> Dict:
 
 def boolean(value: Any) -> bool:
     """Validate and coerce a boolean value."""
+    if value is None:
+        return False
+    if isinstance(value, bool):
+        return value
     if isinstance(value, str):
-        value = value.lower()
+        value = value.lower().strip()
         if value in ('1', 'true', 'yes', 'on', 'enable'):
             return True
         if value in ('0', 'false', 'no', 'off', 'disable'):
             return False
         raise vol.Invalid('invalid boolean value {}'.format(value))
-    return bool(value)
+    if isinstance(value, Number):
+        return value != 0
+    raise vol.Invalid('invalid boolean value {}'.format(value))
+
+
+def boolean_true(value: Any) -> bool:
```

Comment:
I think the changes to the true validator are good, but I don't see what the use case of this is ? 

I also think that if we want to implement it, we should generalize it. Like `with_invalid_value(boolean, False)`, that will return a value if the passed in validator raises Invalid. However, I think that we're missing the point of config validation in that case, we shouldn't accept any value.

Reply:
Makes sense. 

The motivation use case was in wait conditions in templates where we really should be using the validator, rather than just use an ad-hock conception of true that didn't align with this one. I didn't want to be logging a pile of errors though so felt making it tolerate other stuff was sensible. 

I can't think of another situation where invalid output is ignored apart from the boolean case.